### PR TITLE
Update go.yml for Go tests

### PR DESCRIPTION
**Pull request by Brandon-G-Tripp**
Add run of Go tests to workflow file and coverage report. Uploads coverage report to repository for reference.

Patch to add CI check for test coverage at 50% and all tests passing.

### CODE REVIEW

**Review by brianfeister — CHANGES_REQUESTED** [review]
It looks like there's an error in the grep command and the coverage check you added is failing, but not due to threshold

**Comment by brianfeister** [comment]
Also, do we want to be piping these to txt files? I'm not sure what the exact limit is, but GitHub allows for artifacts from test runs but storage is not unlimited

**Comment by brianfeister** [comment]
Sorry, still processing this. Sooooo is exiting `1` expected behavior because that signals that the threshold isn't met? If that's the case I think we want to echo something in the output rather than silently failing without explanation that the coverage threshold is unmet

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -32,3 +33,29 @@ jobs:
...
+
+      - name: Check coverage
+        run: |
+          COVERAGE=$(awk -F'[[:space:]]+' '/^total:/ {print $NF+0}' coverage.txt)
+          echo "Total coverage: $COVERAGE%"
+          if (( $(echo "$COVERAGE < 50" | bc -l) )); then
+            echo "🔴 Coverage is $COVERAGE%. We require minimum 50%."  
+            exit 1
+          elif (( $(echo "$COVERAGE < 60" | bc -l) )); then
+            echo "🟠 Coverage is $COVERAGE%. It's passing but is close to the failure cutoff."
```
Comment (msnorth) [inline .github/workflows/go.yml:51]: I think Java had some sort of utility we used which made sure our coverage never decreased. This is good for now but something to think of

**Review by msnorth — APPROVED** [review]

**Review by brianfeister — APPROVED** [review]

**Review by brianfeister — APPROVED** [review]

**Pull request merged**